src/util/vec2d.rs

```diff
@@ -0,0 +1,137 @@
+use std::ops::{Index, IndexMut};
+
+/// A 2 dimensional matrix in row-major order backed by a contiguous `Vec`
```

Comment:
It's rather backed by a slice now.

Nit: also add a period `.` at the end of the sentence.